Add tests for existing JavaScript code

alex-ju opened the pull request:
In the past few weeks we added quite a bunch of JavaScript code to get features out in the wild and test them during user research and so we sinned by not writing unit tests and rely only on functional tests. This PR is trying to address this issue by adding a set of tests for the existing JavaScript code with the hope to add more tests that would cover the server requests.

Run tests locally with `rake jasmine`

<img width="786" alt="jasmine-2" src="https://user-images.githubusercontent.com/788096/47430390-0a6dd500-d791-11e8-8ac0-7b803805124d.png">

[Trello card](https://trello.com/c/0AXSQGEx)

alex-ju commented on spec/javascripts/components/error-alert-spec.js:
```diff
@@ -0,0 +1,30 @@
+
+  it('should be focused', function () {
+    module.focus()
+    expect(element[0]).toBeFocused()
```
Yes, it should. But we want to ensure the state not the style.

benthorner replied:
👍 Just wanted to check you'd seen it.

alex-ju commented on spec/javascripts/components/markdown-editor-spec.js:
```diff
@@ -0,0 +1,150 @@
+    })
+
+    it('should display the editor head', function () {
+      var head = $('.js-markdown-editor__head')
```
Not my choice of naming. Would _should display the editor header_ be more decent?

benthorner replied:
Ah, I see. Using 'header' would make more sense, although I wonder if this is actually covered by the clicking buttons specs below? Would they pass if this spec failed?

alex-ju commented on spec/javascripts/helpers/jasmine_examples/SpecHelper.js:
```diff
@@ -0,0 +1,13 @@
+beforeEach(function () {
+  jasmine.addMatchers({
+    toBeFocused: function () {
```
We can use [jasmine-jquery](https://github.com/velesin/jasmine-jquery) to be consistent with the govuk_publishing_components as we need to be able to move these components with tests upstream.

alex-ju replied:
Sounds good. Will do that.

alex-ju commented:
@benthorner addressed your comments (including redundant tests and nesting). Regarding the matchers we don't have much flexibility as we need to be able to move them in the shared library.
